cylc/flow/templatevars.py

```diff
@@ -17,11 +17,39 @@
 
 from ast import literal_eval
 from optparse import Values
+from pathlib import Path
 from typing import Any, Dict
 
 from cylc.flow.exceptions import InputError
 
 
+from cylc.flow.rundb import CylcWorkflowDAO
+
+
+class OldTemplateVars:
+    """Gets template variables stored in workflow database.
+
+    Mirrors the interface used in scheduler.py to get db info on restart.
+    """
+    DB = 'log/db'
+
+    def __init__(self, run_dir):
+        self.template_vars = {}
+        self.run_dir = Path(run_dir)
+        self._get_db_template_vars()
+
+    def _callback(self, _, row):
+        """Extract key and value and run eval_var on them assigning
+        them to self.template_vars.
+        """
+        self.template_vars[row[0]] = eval_var(row[1])
+
+    def _get_db_template_vars(self):
+        if (self.run_dir / self.DB).exists():
+            dao = CylcWorkflowDAO(str(self.run_dir / self.DB))
+            dao.select_workflow_template_vars(self._callback)
+
+
```

Comment:
I think this is another rebase error, this was replaced by `_prepend_old_templatevars`.